services/auth/oauth2.go

```diff
@@ -91,7 +92,18 @@ func (o *OAuth2) userIDFromToken(req *http.Request, store DataStore) int64 {
 	}
 	t, err := auth_model.GetAccessTokenBySHA(tokenSHA)
 	if err != nil {
-		if !auth_model.IsErrAccessTokenNotExist(err) && !auth_model.IsErrAccessTokenEmpty(err) {
+		if auth_model.IsErrAccessTokenNotExist(err) {
+			// check task token
+			task, err := actions_model.GetRunningTaskByToken(db.DefaultContext, tokenSHA)
```

Comment:
Do we need to add task token checking for OAuth2?

Reply:
It is necessary now, because "Basic Authorization" reads authorization header as `Authorization: basic xxx`

https://github.com/go-gitea/gitea/blob/a35749893b91db48310d91ae0a32fee3ad3bb901/services/auth/basic.go#L49-L57

But some actions like https://github.com/peter-evans/create-or-update-comment send token as: `Authorization: token xxx`, this is what I caught:

![image](https://user-images.githubusercontent.com/9418365/209919185-0ba54fb1-4a42-4c34-91f0-317a8c7a998b.png)